Build & release arm64

carlcsaposs-canonical opened the pull request.

carlcsaposs-canonical committed:
Use reusable workflows

carlcsaposs-canonical committed:
comments

carlcsaposs-canonical committed:
arm64

taurus-forever reviewed (approved):
Update README (same as snap) and please create backlog ticket to add tests here.

carlcsaposs-canonical committed:
Remove override-stage

Current `override-stage` has no effect on final rock

I inspected the rock we're currently building (i.e. ghcr.io/canonical/charmed-mysql:8.0.37-22.04_edge), and it looks like it doesn't contain `/var/lib/mysql`
```
root@foo-0:/# find / -name mysql
/home/mysql
/usr/lib/x86_64-linux-gnu/perl5/5.34/DBD/mysql
/usr/lib/x86_64-linux-gnu/perl5/5.34/auto/DBD/mysql
/usr/lib/mysql
/usr/share/mysql
/usr/bin/mysql
/var/log/mysql
/etc/init.d/mysql
/etc/mysql
root@foo-0:/# mysqld --initialize-insecure -u mysql
root@foo-0:/# find / -name mysql
/home/mysql
/usr/lib/x86_64-linux-gnu/perl5/5.34/DBD/mysql
/usr/lib/x86_64-linux-gnu/perl5/5.34/auto/DBD/mysql
/usr/lib/mysql
/usr/share/mysql
/usr/bin/mysql
/var/log/mysql
/var/lib/mysql
/var/lib/mysql/mysql
/etc/init.d/mysql
/etc/mysql
```

so I think the entire `override-stage` in our rock is doing nothing

carlcsaposs-canonical committed:
update readme

carlcsaposs-canonical commented on rockcraft.yaml:
```diff
@@ -32,30 +33,6 @@ parts:
             - python3-yaml
         stage-snaps:
             - charmed-mysql/8.0/edge
-        override-stage: |
-            LIB_DIR=$CRAFT_STAGE/usr/lib/x86_64-linux-gnu
-            craftctl default
-            mkdir -p $CRAFT_STAGE/var/lib/mysql
-            ln -sf $LIB_DIR/libnuma.so.1 \
-                /usr/lib/x86_64-linux-gnu/libnuma.so.1
-            ln -sf $LIB_DIR/libaio.so.1 \
-                /usr/lib/x86_64-linux-gnu/libaio.so.1
-            ln -sf $LIB_DIR/libicuuc.so.70 \
-                /usr/lib/x86_64-linux-gnu/libicuuc.so.70
-            ln -sf $LIB_DIR/libicui18n.so.70 \
-                /usr/lib/x86_64-linux-gnu/libicui18n.so.70
-            ln -sf $LIB_DIR/libevent_core-2.1.so.7 \
-                /usr/lib/x86_64-linux-gnu/libevent_core-2.1.so.7
-            ln -sf $LIB_DIR/libevent_pthreads-2.1.so.7 \
-                /usr/lib/x86_64-linux-gnu/libevent_pthreads-2.1.so.7
-            ln -sf $LIB_DIR/libprotobuf-lite.so.23 \
-                /usr/lib/x86_64-linux-gnu/libprotobuf-lite.so.23
-            ln -sf $LIB_DIR/libicudata.so.70 \
-                /usr/lib/x86_64-linux-gnu/libicudata.so.70
-            rm -rf /var/lib/mysql/
-            mysqld --initialize
-            cp -r /var/lib/mysql $CRAFT_STAGE/var/lib
-            chown 584788:584788 $CRAFT_STAGE/var/lib/mysql
```
I inspected the rock we're currently building (i.e. ghcr.io/canonical/charmed-mysql:8.0.37-22.04_edge), and it looks like it doesn't contain `/var/lib/mysql`
```
root@foo-0:/# find / -name mysql
/home/mysql
/usr/lib/x86_64-linux-gnu/perl5/5.34/DBD/mysql
/usr/lib/x86_64-linux-gnu/perl5/5.34/auto/DBD/mysql
/usr/lib/mysql
/usr/share/mysql
/usr/bin/mysql
/var/log/mysql
/etc/init.d/mysql
/etc/mysql
root@foo-0:/# mysqld --initialize-insecure -u mysql
root@foo-0:/# find / -name mysql
/home/mysql
/usr/lib/x86_64-linux-gnu/perl5/5.34/DBD/mysql
/usr/lib/x86_64-linux-gnu/perl5/5.34/auto/DBD/mysql
/usr/lib/mysql
/usr/share/mysql
/usr/bin/mysql
/var/log/mysql
/var/lib/mysql
/var/lib/mysql/mysql
/etc/init.d/mysql
/etc/mysql
```

so I think the entire `override-stage` in our rock is doing nothing

paulomach commented on .github/workflows/release.yaml:
```diff
@@ -0,0 +1,62 @@
+          pattern: ${{ needs.build.outputs.artifact-prefix }}-*
+          merge-multiple: true
+      - name: Generate SBOM(s)
+        shell: python
+        run: |
+          import pathlib
+          import subprocess
+
+          for rock_file in pathlib.Path(".").glob("*.rock"):
+              subprocess.run(
+                  ["syft", rock_file.name, "--output", f"spdx-json={rock_file.name}.spdx.json"],
+                  check=True,
+              )
```
good ol shell?

```suggestion
        shell: bash
        run: |
          for rock_file in *.rock
          do
            syft "${rock_file}" --output spdx-json="${rock_file}".spdx.json
          done    
```

carlcsaposs-canonical replied:
it'll probably move to reusable workflows at some point

to keep consistent with postgres repo, not changing for now

paulomach reviewed (approved):
good cleanup
